pepper-apis/src/main/java/org/broadinstitute/ddp/db/ActivityInstanceDao.java

```diff
@@ -378,35 +380,60 @@ public Long getActivityInstanceIdByGuid(Handle handle, String guid) {
+     * @param handle    the database handle
+     * @param userId    the user who owns the activity instances
      * @param summaries the list of summaries
      */
-    public void renderActivitySummaryText(Handle handle, long userId, List<ActivityInstanceSummary> summaries) {
-        if (!summaries.isEmpty()) {
-            Set<Long> instanceIds = summaries.stream().map(ActivityInstanceSummary::getActivityInstanceId).collect(Collectors.toSet());
-            var instanceDao = handle.attach(org.broadinstitute.ddp.db.dao.ActivityInstanceDao.class);
-            var substitutions = instanceDao.bulkFindSubstitutions(instanceIds)
-                    .collect(Collectors.toMap(wrapper -> wrapper.getActivityInstanceId(), wrapper -> wrapper.unwrap()));
-            var sharedSnapshot = I18nContentRenderer
-                    .newValueProviderBuilder(handle, userId)
-                    .build().getSnapshot();
-            for (var summary : summaries) {
-                if (StringUtils.isBlank(summary.getActivitySummary())) {
-                    continue;
+    public void renderActivitySummary(Handle handle, long userId, List<ActivityInstanceSummary> summaries) {
+        if (summaries.isEmpty()) {
+            return;
+        }
+
+        Set<Long> instanceIds = summaries.stream().map(ActivityInstanceSummary::getActivityInstanceId).collect(Collectors.toSet());
+        var instanceDao = handle.attach(org.broadinstitute.ddp.db.dao.ActivityInstanceDao.class);
+        var substitutions = instanceDao.bulkFindSubstitutions(instanceIds)
+                .collect(Collectors.toMap(wrapper -> wrapper.getActivityInstanceId(), wrapper -> wrapper.unwrap()));
+        var sharedSnapshot = I18nContentRenderer
+                .newValueProviderBuilder(handle, userId)
+                .build().getSnapshot();
+
+        for (var summary : summaries) {
+            if (StringUtils.isBlank(summary.getActivitySummary())) {
+                continue;
+            }
+
+            Map<String, String> subs = substitutions.getOrDefault(summary.getActivityInstanceId(), new HashMap<>());
+            var provider = new RenderValueProvider.Builder()
+                    .setActivityInstanceNumber(summary.getInstanceNumber())
+                    .withSnapshot(sharedSnapshot)
+                    .withSnapshot(subs)
+                    .build();
+            Map<String, Object> context = new HashMap<>();
+            context.put(I18nTemplateConstants.DDP, provider);
+
+            // Render the name.
+            final String nameText;
+            if (summary.getInstanceNumber() > 1) {
+                if (StringUtils.isNotBlank(summary.getActivitySecondName())) {
+                    nameText = renderer.renderToString(summary.getActivitySecondName(), context);
+                } else {
+                    String originalName = renderer.renderToString(summary.getActivityName(), context);
+                    nameText = originalName + " #" + summary.getInstanceNumber();
```

Comment:
Are we going to have the first second instance have a #2 suffix (we start at 1 right? This might be good to annotate somewhere). Might be a little weird to have a #2 without a #1

Reply:
Actually, it has always been that way. First instance has no number, second instance has `#2`, third instance has `#3`, etc. I'm keeping that behavior so we don't change anything for existing studies.